### PR TITLE
Miscellaneous fixes

### PR DESCRIPTION
**Pull request by Soreepeong**
* Plugin constructors are now ran on separate threads, so that `Task.Result` and `Task.Wait()` inside there will not result in blocking random task thread pool threads.
* Made `InterfaceManager` wait for swap chain before setting title bar color according to dark mode.
* Suppressed ignorable errors from `CurrentProcessModules` and `DalamudAssetManager`.
* Changed `Localization` to a provided service, so that `LoadingDialog` can use `Loc`.
* `LoadingDialog` now does not depends on `System.Windows.Forms`, and instead it calls the underlying functions directly. This requires setting the activation context, which is not exposed by `System.Windows.Forms`. This suppresses thread association errors that are safely suppress-and-continuable.
* Made console window display all logs ever since initializing Serilog.
* Made loading dialog show last 20 lines of logs.
image

### CODE REVIEW

File: Dalamud/IoC/Internal/ServiceContainer.cs [hunk 1]
```diff
@@ -127,7 +128,26 @@ public void RegisterInterfaces(Type type)
             return null;
         }
 
-        ctor.Invoke(instance, resolvedParams);
+        var tcs = new TaskCompletionSource(TaskCreationOptions.RunContinuationsAsynchronously);
```
Comment (goaaats) [inline Dalamud/IoC/Internal/ServiceContainer.cs:131]: Could you leave a comment here explaining why this is necessary?

**Review by goaaats — CHANGES_REQUESTED** [review]
Looks good code-wise!

File: Dalamud/IoC/Internal/ServiceContainer.cs (hunk 1 above)
Comment (Soreepeong) [reply Dalamud/IoC/Internal/ServiceContainer.cs:131]: I'm not sure if this fixes problems; more of seeing if this fixes stalling problems on load time.

**Comment by Soreepeong** [comment]
Made all TaskCompletionSource have RunContinuationsAsynchronously passed too.

File: Dalamud/IoC/Internal/ServiceContainer.cs (hunk 1 above)
Comment (goaaats) [reply Dalamud/IoC/Internal/ServiceContainer.cs:131]: Alright, let's leave it like this for now and see if things work out

**Review by goaaats — APPROVED** [review]

**Pull request merged**